04-onion-routing.md

```diff
@@ -873,8 +931,54 @@ An _intermediate hop_ MUST NOT, but the _final node_:
   the final node's HTLC:
     - MUST return `final_incorrect_cltv_expiry` error.
   - if the `amt_to_forward` is greater than the `incoming_htlc_amt` from the
-  final node's HTLC:
+  final node's HTLC, AND `incomplete_payment` is not set:
     - MUST return a `final_incorrect_htlc_amount` error.
+  - if `incomplete_payment` is set:
+    - MUST use `amt_to_forward` to evaluate for
+      `incorrect_payment_amount` errors.
+    - SHOULD suspend incoming HTLCs:
+      - if the incoming HTLC has an `amt_to_forward` that is
+        different from existing suspended payments of the same
+        `payment_hash`:
+        - MUST fail the HTLC.
+        - MUST return an `incorrect_payment_amount` error.
+      - if the total `incoming_htlc_amt` of all suspended HTLCs and
+        the new incoming HTLC equals or exceeds the `amt_to_forward`,
+        of the same `payment_hash`:
+        - MUST succeed the HTLC and all suspended HTLCs.
+      - if a new block arrives while suspended HTLCs exist:
+        - MUST reevaluate the `cltv_expiry` of the suspended HTLCs.
+      - if an HTLC has been suspended for too long:
+        - MUST fail the HTLC.
+        - MUST return an `incomplete_payment_too_slow` error.
+
+#### Rationale
+
+If `incomplete_payment` is set, but the total expected value has not
+arrived at the final node, and the final node claims the payment,
+then the final node has effectively sold the `payment_preimage` at a
+discount.
+
+This is effectively "economically rational atomicity".
+A final node that wants to lose money may voluntarily claim a partial
+payment that is less than its advertised payment amount.
+We assume that no final node would want to lose money.
+
+We assume here that the `payment_preimage` is of value.
```

Comment:
I would leave out this 3rd Paragraph of the 4 added rational paragraphs. I think the other 3 work without this one